connectors/sources/postgresql.py

```diff
@@ -295,21 +299,50 @@ async def data_streamer(self, table=None, query=None):
         Yields:
             list: It will first yield the column names, then data in each row
         """
-        async for data in fetch(
-            cursor_func=partial(
-                self.get_cursor,
-                self.queries.table_data(
-                    schema=self.schema,
-                    table=table,
-                )
-                if query is None
-                else query,
-            ),
-            fetch_columns=True,
-            fetch_size=self.fetch_size,
-            retry_count=self.retry_count,
-        ):
-            yield data
+        if query is None and row_count is not None and order_by_columns is not None:
+            order_by_columns_list = ",".join(order_by_columns)
+            offset = 0
+            fetch_columns = True
+            while True:
+                async for data in fetch(
+                    cursor_func=partial(
+                        self.get_cursor,
```

Comment:
What's the role of the cursor here?

Reply:
So, basically, this is a generic functionality for PostgreSQL, Oracle and MS-SQL. Currently, I do not remove the cursor-based approach. I have divided the query to fetch data into chunks.